### ISSUE TITLE
RE: ROS2 support for HEBI Arm kit

### ISSUE THREAD

**Issue by AllisonSciullo** [AllisonSciullo/support_ticket_test#50]
New email received from 'Norbert Reider' via Support (support@hebirobotics.com) on Tue, 21 Nov 2023 07:12:47 +0000:

Hi Chris,

Thank you so much, now everything works like a charm, the HEBI Arm kit is controlled successfully from ROS2!

Thanks,
Norbert

From: Chris Bollinger <chris@hebirobotics.com>
Sent: Thursday, November 16, 2023 7:22 PM
To: Norbert Reider <norbert.reider@ericsson.com>
Cc: Florian Enner <florian@hebirobotics.com>; Support <support@hebirobotics.com>; Matthew Tesch <matt@hebirobotics.com>
Subject: Re: ROS2 support for HEBI Arm kit

Hi Norbert,

It's public now. I forgot we had to break that out separately for ROS2. I'll keep an eye on the thread in case there are any more I've missed.

Best,

On Thu, Nov 16, 2023 at 2:30 AM Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>> wrote:
Hi Chris,

Great, thank you for your fast response, very much appreciated!
One more thing, there is another dependency “hebi_msgs”, could you share that somehow as well?

package.xml:  <depend>hebi_msgs</depend>

Thank you!
Regards,
Norbert

From: Chris Bollinger <chris@hebirobotics.com<mailto:chris@hebirobotics.com>>
Sent: Wednesday, November 15, 2023 4:34 PM
To: Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>>
Cc: Florian Enner <florian@hebirobotics.com<mailto:florian@hebirobotics.com>>; Support <support@hebirobotics.com<mailto:support@hebirobotics.com>>; Matthew Tesch <matt@hebirobotics.com<mailto:matt@hebirobotics.com>>
Subject: Re: ROS2 support for HEBI Arm kit

Hi Norbert,

Those should be visible to you now, sorry about that.

Best,

On Wed, Nov 15, 2023 at 7:08 AM Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>> wrote:
Hi Chris,

Thank you for your support in advance! I have now another question: There are references in the https://github.com/HebiRobotics/hebi_ros2_examples/blob/main/README.md<https://protect2.fireeye.com/v1/url?k=31323334-501cfaf3-313273af-454445554331-327c03a6d5b1b145&q=1&e=160ad15b-bee0-4536-93cb-85c3a14198e4&u=https%3A%2F%2Fgithub.com%2FHebiRobotics%2Fhebi_ros2_examples%2Fblob%2Fmain%2FREADME.md> repo to two other repos. They seem to be private, could you make them public please to be able to access them?

https://github.com/HebiRobotics/hebi_hardware<https://protect2.fireeye.com/v1/url?k=31323334-501cfaf3-313273af-454445554331-3bd6e2295b54267c&q=1&e=160ad15b-bee0-4536-93cb-85c3a14198e4&u=https%3A%2F%2Fgithub.com%2FHebiRobotics%2Fhebi_hardware>
https://github.com/HebiRobotics/hebi_bringup<https://protect2.fireeye.com/v1/url?k=31323334-501cfaf3-313273af-454445554331-6a476f6fd83a1462&q=1&e=160ad15b-bee0-4536-93cb-85c3a14198e4&u=https%3A%2F%2Fgithub.com%2FHebiRobotics%2Fhebi_bringup>

Thanks,
Norbert

From: Norbert Reider
Sent: Tuesday, October 3, 2023 3:05 PM
To: 'Chris Bollinger' <chris@hebirobotics.com<mailto:chris@hebirobotics.com>>
Subject: RE: ROS2 support for HEBI Arm kit

Great, now it works, thank you! I will get back to you in case there is any question or feedback.

Thanks,
Norbert

From: Chris Bollinger <chris@hebirobotics.com<mailto:chris@hebirobotics.com>>
Sent: Tuesday, October 3, 2023 2:53 PM
To: Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>>
Subject: Re: ROS2 support for HEBI Arm kit

It was private, let me know if you can see it now.

Best,

On Tue, Oct 3, 2023 at 8:46 AM Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>> wrote:
Hi Chris,

Thank you for your reply. The link does not work for me, I cannot find that repo in the list of HEBI repositories (https://github.com/orgs/HebiRobotics/repositories<https://protect2.fireeye.com/v1/url?k=31323334-501cfaf3-313273af-454445554331-be5825031fd11e88&q=1&e=30b19971-63f8-4521-a0c8-dfa96c2385cf&u=https%3A%2F%2Fgithub.com%2Forgs%2FHebiRobotics%2Frepositories>). Maybe it is not yet public.
Please check that if you have time.

Thank you in advance!
Regards,
Norbert

From: Chris Bollinger <chris@hebirobotics.com<mailto:chris@hebirobotics.com>>
Sent: Tuesday, October 3, 2023 1:45 PM
To: Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>>
Cc: Florian Enner <florian@hebirobotics.com<mailto:florian@hebirobotics.com>>; Support <support@hebirobotics.com<mailto:support@hebirobotics.com>>; Matthew Tesch <matt@hebirobotics.com<mailto:matt@hebirobotics.com>>
Subject: Re: ROS2 support for HEBI Arm kit

Hi Norbert,

Apologies for the slow reply, I lost track of this email chain.

We are still working to get the ROS2 port released as official binary packages, but there are a few organizational changes that need to complete before that can happen. While we are waiting for that to go through, there are instructions located at https://github.com/HebiRobotics/hebi_ros2_examples/blob/main/README.md<https://protect2.fireeye.com/v1/url?k=31323334-501d5122-313273af-454445555731-1f161e4d24265e4f&q=1&e=4c4832b8-4b1d-4f0a-8780-9aadbcf07bfc&u=https%3A%2F%2Fgithub.com%2FHebiRobotics%2Fhebi_ros2_examples%2Fblob%2Fmain%2FREADME.md> that can be followed to setup a HEBI arm using the ROS2 stack.

Please let me know if you have any questions or feedback about the way we've set things up, or any needs that aren't covered by what we have developed so far.

Best,
Chris Bollinger

On Mon, Oct 2, 2023 at 9:24 AM Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>> wrote:
Hi Chris,

I hope you are fine. Do you have some update on the status of the ROS2 support?

Thank you in advance!
Regards,
Norbert

From: Florian Enner <florian@hebirobotics.com<mailto:florian@hebirobotics.com>>
Sent: Monday, September 4, 2023 10:32 PM
To: Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>>; Support <support@hebirobotics.com<mailto:support@hebirobotics.com>>; Chris Bollinger <chris@hebirobotics.com<mailto:chris@hebirobotics.com>>; Matthew Tesch <matt@hebirobotics.com<mailto:matt@hebirobotics.com>>
Subject: Re: ROS2 support for HEBI Arm kit

Hi Norbert,

Chris (CC'd) is actively working on official ROS2 support. As far as I know the beta release will be in the near future, but he will be able to tell you more about it.

That looks like a fun project. Good luck and keep us updated :)

Best,
Florian

On Mon, Sep 4, 2023 at 5:38 PM Norbert Reider <norbert.reider@ericsson.com<mailto:norbert.reider@ericsson.com>> wrote:
Hi Florian,

I hope you are doing well.

I would like to ask you about ROS 2 support for a HEBI arm kit. We extended your mecanum drive mobile base platform with an arm kit. The arm kit is a 6-DoF X-Series Arm from your secret customer 😊 (BME university). I am running a joint university project with them and we would like to control the arm in a ROS 2 environment if possible (e.g., with MoveIt2). Please let me know if you have already done this or have some input.

Thank you!
Regards,
Norbert
